internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -925,13 +924,19 @@ func (c *Coordinator) DiagnosticHooks() diagnostics.Hooks {
 				// In this case, the state is re-fetched to obtain the latest
 				// information. After the second fetch, no more checks are
 				// performed and the state is returned as is.
-				for _, cs := range compStates {
-					if cs.State.State == client.UnitStateHealthy ||
-						(cs.State.VersionInfo.BuildHash == "" ||
-							cs.State.VersionInfo.Meta["commit"] == "") {
-						s, compStates = getState(c)
-						break
+				var s State
+				var compStates []StateComponentOutput
+			outerLoop:
+				for i := 0; i < 5; i++ {
+					s, compStates = getState(c)
+					for _, cs := range compStates {
+						if cs.State.State == client.UnitStateHealthy &&
+							cs.State.VersionInfo.BuildHash != "" &&
+							cs.State.VersionInfo.Meta["commit"] != "" {
+							break outerLoop
+						}
```

Comment:
This whole block would be much better as its own function either in `getState` or another wrapper around it.

I still am not a fan of this, again your just improving the odds and not solving the true issue. Could we some how guarantee that when status is updated to Healthy that the VersionInfo is set at the same time? Why are they not?

If there is a bad component that never provides the versioninfo it is going to delay this result as well indefinitely by 250 milliseconds.